app/models/submission/validations_by_template_name.rb

```diff
@@ -65,5 +80,94 @@ def validate_consistent_column_value(column_header)
       )
     end
   end
-  # rubocop:enable Metrics/MethodLength,Metrics/AbcSize
+  # rubocop:enable Metrics/MethodLength
+
+  def validate_samples_per_pool_for_labware
+    return if headers.index(HEADER_BARCODE).nil? && headers.index(HEADER_PLATE_WELLS).nil?
+
+    grouped_rows = group_rows_by_study_and_project
+    grouped_rows.each_value { |rows| process_rows(rows) }
+  end
+
+  private
+
+  def process_rows(rows)
+    barcodes = rows.pluck(headers.index(HEADER_BARCODE))
+    well_locations = rows.pluck(headers.index(HEADER_PLATE_WELLS))
+
+    if plate?(barcodes, well_locations)
+      validate_for_plates(barcodes, well_locations, rows)
+    elsif tube?(barcodes, well_locations)
+      validate_for_tubes(barcodes, rows)
+    end
+  end
+
+  # rubocop:disable Metrics/AbcSize
+  def validate_for_plates(barcodes, well_locations, rows)
+    plate = Plate.find_from_any_barcode(barcodes.uniq.first)
+    return if plate.nil?
+
+    wells = plate.wells.for_bulk_submission.located_at(well_locations)
+    total_number_of_samples_per_study_project = wells.map(&:samples).flatten.count.to_i
+    number_of_pools = rows.pluck(headers.index(HEADER_NUMBER_OF_POOLS)).uniq.first.to_i
+
+    validate_samples_per_pool(rows, total_number_of_samples_per_study_project, number_of_pools)
+  end
+  # rubocop:enable Metrics/AbcSize
+
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable Metrics/MethodLength
+  def validate_for_tubes(barcodes, rows)
```

Comment:
refactor this? rubocop clearly not happy
ask codepilot for ideas?
plus add more test coverage